x-pack/plugins/apm/public/components/app/Settings/anomaly_detection/index.tsx

```diff
@@ -7,11 +7,17 @@
 import React, { useState } from 'react';
 import { EuiTitle, EuiSpacer, EuiText } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import { EuiPanel } from '@elastic/eui';
 import { JobsList } from './jobs_list';
 import { AddEnvironments } from './add_environments';
 import { useFetcher, FETCH_STATUS } from '../../../../hooks/useFetcher';
+import { LicensePrompt } from '../../../shared/LicensePrompt';
+import { useLicense } from '../../../../hooks/useLicense';
 
 export const AnomalyDetection = () => {
+  const license = useLicense();
+  const hasValidLicense = license?.isActive && license?.hasAtLeast('platinum');
```

Comment:
We have this method in service maps, maybe we should put it in some utils: https://github.com/elastic/kibana/blob/f5b77e1026667e2b2978ea114e59f1227d25cdc8/x-pack/plugins/apm/common/service_map.ts#L45-L47

Reply:
imo that's a slight overabstraction of something that's pretty trivial in the first place.
If this logic grows I think it could make sense.